src/goose/batch.clj

```diff
@@ -1,9 +1,22 @@
-(ns goose.batch)
+(ns goose.batch
+  (:require
+    [goose.utils :as u]))
+
+(def status-in-progress "in-progress")
+(def status-complete "complete")
 
 (defn new
   [{:keys [callback-fn-sym]} jobs]
   (let [id (str (random-uuid))]
     {:id              id
      :callback-fn-sym callback-fn-sym
-     :jobs            (map #(assoc % :batch-id id) jobs)}))
+     :jobs            (map #(assoc % :batch-id id) jobs)
+     :created-at      (u/epoch-time-ms)}))
+
+(defn status-from-counts
+  [{:keys [successful dead]}
+   total]
+  (cond
+    (= (+ successful dead) total) status-complete
```

Comment:
Are we reporting job status as complete even if all are dead?
I was thinking 3 complete status: successful, dead, partially-successful.

Reply:
These statuses, `successful`, `dead`, `partially-successful`, might be redundant since we can derive them from the counts.

We decided to keep this as is for now and update it after further discussion, if we need to.
Resolving this conversation and moving it to Slack.